test(formatter): run conformance tests with different options

Dunqing opened the pull request:
Context: https://github.com/oxc-project/oxc/issues/17151#issuecomment-3680097759

We have to run conformance tests with different options from the defaults, especially those options (semicolon, quote props) that can cause a syntax error if there is a bug

Boshen reviewed (commented):
Very nice 👍

leaysgur reviewed (approved):
Anyway, let's merge first.

leaysgur commented:
### Merge activity

* **Dec 24, 1:04 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing commented on tasks/coverage/src/tools/formatter.rs:
```diff
@@ -15,10 +19,43 @@ use crate::{
+        FormatOptions {
+            indent_width: IndentWidth::try_from(8).unwrap(),
+            line_width: LineWidth::try_from(120).unwrap(),
+            line_ending: LineEnding::Lf,
```
It should be removed, actually, since most of the link break checkers in the formatter don't care about `\r`. That said, we don't guarantee `LineEnding::Cr` would work well.

Dunqing committed:
test(formatter): run conformance tests with different options (#17261)

Context: https://github.com/oxc-project/oxc/issues/17151#issuecomment-3680097759

We have to run conformance tests with different options from the defaults, especially those options (semicolon, quote props) that can cause a syntax error if there is a bug
